install.ps1

```diff
@@ -1,429 +1,888 @@
+        Write-Host "[-] Do you still wish to proceed? (Y/N): " -ForegroundColor Yellow -NoNewline
+        $response = Read-Host
+        if ($response -notin @("y","Y")) {
+            exit 1
+        }
     } else {
-      iex ((New-Object System.Net.WebClient).DownloadString('https://boxstarter.org/bootstrapper.ps1')); Get-Boxstarter -Force
+        Write-Host "`t[+] Disk is larger than 60 GB" -ForegroundColor Green
     }
-    return $true
-  } catch {
-    Write-Host "Failed to install boxstarter. Trying again."
-  }
-
-  # https://stackoverflow.com/questions/11696944/powershell-v3-invoke-webrequest-https-error
-  # Allows current PowerShell session to trust all certificates
-  # Also a good find: https://www.briantist.com/errors/could-not-establish-trust-relationship-for-the-ssltls-secure-channel/
-  try {
-  Add-Type @"
-  using System.Net;
-  using System.Security.Cryptography.X509Certificates;
-  public class TrustAllCertsPolicy : ICertificatePolicy {
-    public bool CheckValidationResult(
-      ServicePoint srvPoint, X509Certificate certificate,
-      WebRequest request, int certificateProblem) {
-      return true;
+
+    # Check if system is a virtual machine
+    $virtualModels = @('VirtualBox', 'VMware Virtual Platform', 'Virtual Machine')
+    if ((Get-WmiObject win32_computersystem).model -notin $virtualModels) {
+        Write-Host "`t[!] You are not on a virual machine or have hardened your machine to not appear as a virtual machine" -ForegroundColor Red
+        Write-Host "`t[!] Please do NOT install this on your host system as it can't be uninstalled completely" -ForegroundColor Red
+        Write-Host "`t[!] ** Please only install on a virtual machine **" -ForegroundColor Red
+        Write-Host "`t[!] ** Only continue if you know what you are doing! **" -ForegroundColor Red
+        Write-Host "[-] Do you still wish to proceed? (Y/N): " -ForegroundColor Yellow -NoNewline
+        $response = Read-Host
+        if ($response -notin @("y","Y")) {
+            exit 1
+        }
     }
-  }
-"@
-  } catch {
-      Write-Host "Failed to add new type"
-  }
-
-  try {
-    # Become overly trusting
-    $prevCertPolicy = [System.Net.ServicePointManager]::CertificatePolicy
-    [System.Net.ServicePointManager]::CertificatePolicy = New-Object TrustAllCertsPolicy
-
-    # Download and install boxstarter
-    if ($PSVersionTable -And $PSVersionTable.PSVersion.Major -ge 5) {
-      . { iwr -useb https://boxstarter.org/bootstrapper.ps1 } | iex; Get-Boxstarter -Force
-    } else {
-      iex ((New-Object System.Net.WebClient).DownloadString('https://boxstarter.org/bootstrapper.ps1')); Get-Boxstarter -Force
+
+    # Prompt user to remind them to take a snapshot
+    Write-Host "[-] Do you need to take a VM snapshot before continuing? (Y/N): " -ForegroundColor Yellow -NoNewline
+    $response = Read-Host
+    if ($response -notin @("n","N")) {
+        exit 1
```

Comment:
For consistency I think we should also make this so `Y` continues and `N` exits.